gke.tf

```diff
@@ -0,0 +1,263 @@
+  ip_range_services          = "${var.infra_name}-gke-service-subnet"
+  http_load_balancing        = false
+  horizontal_pod_autoscaling = true
+  network_policy             = false
+# remove_default_node_pool   = true
+#  initial_node_count         = 1
+  create_service_account     = true
+  grant_registry_access      = true
+  
+
+  node_pools = [
+    {
+      name                   = "linux-pool"
+      machine_type           = "e2-highcpu-2"
+      min_count              = 1
+      max_count              = 8
+      local_ssd_count        = 0
+      disk_size_gb           = 100
+      disk_type              = "pd-standard"
+      image_type             = "UBUNTU" // 20.04
+      auto_repair            = false
+      auto_upgrade           = false
+      preemptible            = false
+      initial_node_count     = 1
+    }     
+  ] 
+
+  node_pools_oauth_scopes = {
+    linux-pool          = ["https://www.googleapis.com/auth/cloud-platform"]
+  }
+
+  node_pools_labels = {
+    all          = {}
+  }
+  node_pools_metadata = {
+    all = {
+      disable-legacy-endpoints = true
+    }
+  }
+  node_pools_taints = {
+    all          = []
+  }
+  node_pools_tags = {
+    all = []
+  }
+}
+
+
+## K8s windows node pool
+
+resource "google_container_node_pool" "windows-pool" {
+  cluster             = module.gke.cluster_id
+  initial_node_count  = 1
+  location            = "${var.region}-${var.zone}"
+  max_pods_per_node   = 10
+  name                = "windows-pool"
+  #node_count          = 0
+  node_locations      = ["${var.region}-${var.zone}"]
+  provider            = google.offensive-pipeline
+  version             = "1.20.9-gke.1001" #Make upgrades from here.
+  autoscaling {
+      max_node_count = 8
+      min_node_count = 1 # at least 1 required since Gitlab K8s runner for windows has issue with scaling-up from 0->1 nodes as node-pool label cant contain the windows build version
+    }
+
+  management {
+      auto_repair  = false
+      auto_upgrade = false
+    }
+
+  node_config {
+      disk_size_gb      = 200
+      disk_type         = "pd-standard"
+      guest_accelerator = []
+      image_type        = "WINDOWS_LTSC"
+      labels            = {
+          "cluster_name" = module.gke.name
+          "node_pool"    = "windows-pool"
+        }
+      local_ssd_count   = 0
+      machine_type      = "n1-standard-2"
+      metadata          = {
+          "cluster_name"               = module.gke.name
+          "disable-legacy-endpoints"   = "true"
+          "node_pool"                  = "windows-pool"
+          "windows"                    = "true"
+          "windows-startup-script-url" = join("", [
+                                              "gs://", google_storage_bucket.deployment_utils.name,
+                                              "/",
+                                              google_storage_bucket_object.disable_windows_defender_ps.name
+                                              ])
```

Comment:
How about declaring this in locals and use reference.

Reply:
Take a look here:
https://github.com/SygniaLabs/ScallOps/commit/68b25391f48853d3f35a286fffd747d6fdf0e15d#diff-9a576c4980b4f226b7ef1042670d564777bbb0ed8ae2d1df33afbd03e2991fcdR6

and here
https://github.com/SygniaLabs/ScallOps/commit/68b25391f48853d3f35a286fffd747d6fdf0e15d#diff-b07c642f04e91b8316f85904eb364d545ed655c969f8c92b54bd7b44b931ff5bL180